src/lib/db/clients/sqlserver.js

```diff
@@ -446,24 +446,54 @@ export async function getPrimaryKey(conn, database, table, schema) {
   return data.recordset && data.recordset[0] && data.recordset.length === 1 ? data.recordset[0].COLUMN_NAME : null
 }
 
-export async function updateValues(conn, updates) {
+export async function applyChanges(conn, changes) {
+  let results = []
 
-  const { updateQueries, selectQueries } = buildUpdateAndSelectQueries(knex, updates)
-  const sql = ['set xact_abort on', 'BEGIN TRANSACTION', ...updateQueries, 'COMMIT'].join(";")
-  const results = []
   await runWithConnection(conn, async (connection) => {
     const cli = { connection }
-    await driverExecuteQuery(cli, { query: sql })
+    await driverExecuteQuery(cli, { query: 'set xact_abort on; BEGIN TRANSACTION' })
```

Comment:
so yes there is this limitation with SQL Server. You'll have to compile them into a single statement, similar to how the old `updateValues` worked:

const sql = ['BEGIN TRANSACTION', ...updateQueries, ...deleteQueries, 'END TRANSACTION']

Reply:
Ah, that's what I was afraid of. I made the necessary changes.